### PR TITLE
TDT compute timestamps option and Extra Whitespace handling for SPE

### PR DESCRIPTION
**Pull request by monica-sekoyan**
# What does this PR do ?

This PR includes option for getting timestamps from TDT Decoder head, including on a segment level. This option is enabled for RNNT and CTC decodings as well. And also introduces workaround for getting tokens and id_s for extra whitespaces (even if the tokenizer was not trained in this way.) This is crucial for getting alignments in the future.

**Collection**: asr, common

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* To get timestamps from TDT decoding head, just change the decoding config of the model in the following way:

```python
import nemo.collections.asr.models as nemo_asr
import omegaconf

tdt_asr_model = nemo_asr.EncDecRNNTBPEModel.from_pretrained(model_name='nvidia/parakeet-tdt_ctc-1.1b')

cfg = model.cfg.decoding
with omegaconf.open_dict(cfg):
    cfg['compute_timestamps']=True

tdt_asr_model.change_decoding_strategy(decoding_cfg=cfg)
```

* To control timestamp level manipulate the following field of decoding config:

```python

cfg = model.cfg.decoding
with omegaconf.open_dict(cfg):
    cfg['compute_timestamps']=True
    cfg['rnnt_timestamp_type'] = 'all' # can be "char", "word", "segment" or "all"

tdt_asr_model.change_decoding_strategy(decoding_cfg=cfg)
```

* To make the SPE tokenizer (already trained) encode the text not ignoring occurring extra whitespaces (which is the default behaviour), modify the tokenizer in the following way:

```python
model.tokenizer.ignore_extra_whitespaces = False
```

* To train the SPE tokenizer to make it ignore extra whitespaces both during training and further encoding, use this flag in the training script:

```shell
python scripts/tokenizers/process_asr_text_tokenizer.py --tokenizer=spe --spe_remove_extra_whitespaces
```

# GitHub Actions CI

The Jenkins CI system has been replaced by GitHub Actions self-hosted runners.

The GitHub Actions CI will run automatically when the "Run CICD" label is added to the PR.
To re-run CI remove and add the label again.
To run CI on an untrusted fork, a NeMo user with write access must first click "Approve and run".

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [x] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: nemo/collections/asr/parts/submodules/rnnt_decoding.py [hunk 1]
```diff
@@ -1006,6 +1045,60 @@ def _get_word_offsets_subwords_sentencepiece(
 
         return word_offsets
 
+    @staticmethod
+    def _get_segment_offsets(
```
Comment (nithinraok) [inline nemo/collections/asr/parts/submodules/rnnt_decoding.py:1049]: can you add the same for `ctc_decoding.py`

File: scripts/tokenizers/process_asr_text_tokenizer.py [hunk 2]
```diff
@@ -137,6 +137,11 @@
     '--spe_control_symbols', default=None, type=str, nargs='+', help='Control symbols for SentencePiece'
 )
 parser.add_argument('--spe_split_digits', action='store_true', help='Split digits into separate tokens.')
+parser.add_argument(
+    '--spe_remove_extra_whitespaces',
```
Comment (nithinraok) [inline scripts/tokenizers/process_asr_text_tokenizer.py:141]: make default value as true

File: nemo/collections/asr/parts/submodules/rnnt_decoding.py [hunk 3]
```diff
@@ -257,12 +264,15 @@ def __init__(self, decoding_cfg, decoder, joint, blank_id: int):
 
         # initialize confidence-related fields
         self._init_confidence(self.cfg.get('confidence_cfg', None))
+        # defining compute_timestamps function
```
Comment (nithinraok) [inline nemo/collections/asr/parts/submodules/rnnt_decoding.py:267]: remove this

File: nemo/collections/asr/parts/submodules/rnnt_decoding.py (hunk 1 above)
Comment (monica-sekoyan) [reply nemo/collections/asr/parts/submodules/rnnt_decoding.py:1088]: done

File: scripts/tokenizers/process_asr_text_tokenizer.py (hunk 2 above)
Comment (monica-sekoyan) [reply scripts/tokenizers/process_asr_text_tokenizer.py:141]: as discussed, default this option is false, if one needs to remove_extra_spaces both during training and inference, they will need to specify this flag

**Pull request closed without merging**